R/subtotals-and-headings.R

```diff
@@ -218,18 +218,10 @@ setMethod("subtotals<-", c("CrunchVariable", "ANY"), function (x, value) {
                                        makeInsertion,
                                        var_categories = categories(x)))
 
-    # grab the old inserts so they are not deleted, but modify the list so we
-    # don't have *too* many extras
-    old_inserts <- Insertions(data = lapply(transforms(x)$insertions,
-                                            makeInsertion,
-                                            var_categories = categories(x)))
-    if (!is.null(old_inserts)) {
-        inserts <- modifyCats(old_inserts, inserts)
-    }
-
-    bd <- wrapEntity("view" = list("transform" = list("insertions" = inserts)))
-    crPATCH(self(x), body=toJSON(bd))
     dropCache(cubeURL(x))
+    bd <- list("transform" = list("insertions" = inserts))
+    # setEntitySlot manages old inserts so they are not duplicated
+    ent <- setEntitySlot(entity(x), "view", bd)
```

Comment:
I'd put the `dropCache` statements after doing the PATCH (`setEntitySlot`)

Reply:
That makes sense (and my first intuition as well), I was modeling this after https://github.com/Crunch-io/rcrunch/blob/master/R/variable.R#L115-L116, which I've changed to drop the cache after as well.

Reply:
Good call, thanks.